libcalico-go/lib/backend/k8s/resources/resources.go

```diff
@@ -138,7 +140,6 @@ func ConvertCalicoResourceToK8sResource(resIn Resource) (Resource, error) {
 	// the metadata annotation.
 	romCopy := &metav1.ObjectMeta{}
 	rom.(*metav1.ObjectMeta).DeepCopyInto(romCopy)
-	romCopy.Name = ""
```

Comment:
I think we probably want to only store the name in the annotation for NetworkPolicy / GlobalNetworkPolicy? i.e., objects that belong to a Tier.